[FLINK-29148][docs][sql-gateway][hive] Add SQL Gateway docs (#20719)
apache/flink: docs/content/docs/dev/table/hiveCompatibility/hiveserver2.md

            <td>Duration</td>
            <td>Keepalive time for an idle worker thread. When the number of workers exceeds min workers, excessive threads are killed after this time interval.</td>
        </tr>
        <tr>
            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.worker.threads.max</h5></td>
            <td>optional</td>
            <td style="word-wrap: break-word;">512</td>
            <td>Integer</td>
            <td>The maximum number of Thrift worker threads</td>
        </tr>
        <tr>
            <td><h5>sql-gateway.endpoint.hiveserver2.thrift.worker.threads.min</h5></td>
            <td>optional</td>
            <td style="word-wrap: break-word;">5</td>
            <td>Integer</td>
            <td>The minimum number of Thrift worker threads</td>
        </tr>
    </tbody>
</table>

Features
----------------

### Built-in HiveCatalog and Hive Dialect

The SQL Gateway with HiveServer2 endpoint aims to provide the same experience compared to the HiveServer2. When users connect to the HiveServer2,
the HiveServer2 endpoint creates the Hive Catalog as the default catalog, switches to the Hive dialect, and executes the SQL in batch mode for the session.

> Users can submit the Hive SQL in Hive style but execute it in the Flink environment.

Reviewer: We can remove the "Features" section and make this as an H2 title. Please mention all the changed default behavior of configuration in this section, e.g., dialect, batch mode, dml-sync, tableau result.

> ### Integrate into the Hive Ecosystem

Reviewer: Rename the section name to "Clients" and use H2 titles to display the supported clients in TOC.

> The HiveServer2 endpoint extends the HiveServer2 API. Therefore, the tools that manage the Hive SQL also work for

Reviewer: ```suggestion
The HiveServer2 Endpoint is compatible with the HiveServer2 wire protocol. Therefore, the tools that manage the Hive SQL also work for
```

> the SQL Gateway with the HiveServer2 endpoint. Currently, Hive JDBC, Hive Beeline, and Dbeaver can connect to the

Reviewer: ```suggestion
the SQL Gateway with the HiveServer2 endpoint. Currently, Hive JDBC, Hive Beeline, Dbeaver, Apache Superset and so on are tested to be able to connect to the
```
